go/vt/vtgate/engine/route.go

```diff
@@ -158,6 +158,9 @@ func (route *Route) RouteType() string {
 
 // GetKeyspaceName specifies the Keyspace that this primitive routes to.
 func (route *Route) GetKeyspaceName() string {
+	if route.Keyspace == nil {
+		return ""
+	}
 	return route.Keyspace.Name
```

Comment:
in which case we will not have a keyspace for a route?

Reply:
removed this code